docs/custom/error-handlers.mdx

```diff
@@ -0,0 +1,154 @@
+---
+id: error_handlers
+title: Custom error handlers
+---
+
+:::caution
+This AMF functionality is currently a work in progress. The following features are being developed:
+- Support for custom error handling in parsing, validation and emission stages
+- Javascript support
+:::
+
+In this documentation you'll learn what is an error handler in AMF and how to create one yourself with code examples.
+
+
+# What's an AMF error handler
```

Comment:
## AMF error types

There are three types of errors in AMF, denoted by the `amf.core.validation.SeverityLevels` object:

- **Info**
  - Indicates something may or may not be wrong
  - Typically informational depending on user intent 
- **Warning**
  - Indicates that something is not correct and action is required
  - Usually warns about future changes, tool deprecations, or errors that will become violations in the future
- **Violation**
  - Indicates that something is in violation of normal operations or is not compliant with its specification
  - Invalidates the AMF graph model invalid